examples/pi-home-router/src/arp/arp_table.rs

```diff
@@ -0,0 +1,61 @@
+use crate::arp::{ipv4_array, ipv6_array};
+use route_rs_packets::MacAddr;
+use std::collections::HashMap;
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+pub(crate) struct ArpTable {
+    /// Should we have separate translation tables for different layer 3 protocols so keys have a fixed size?
+    // Mappings between ((protocol type, protocol address) -> 48-bit Ethernet address)
+    ipv4_mac_translations: HashMap<Ipv4Addr, MacAddr>,
+    ipv6_mac_translations: HashMap<Ipv6Addr, MacAddr>,
+}
+
+impl ArpTable {
+    pub fn new() -> Self {
+        ArpTable {
+            ipv4_mac_translations: HashMap::new(),
+            ipv6_mac_translations: HashMap::new(),
+        }
+    }
+
+    pub fn contains_key(&self, protocol_type: u16, protocol_address: &[u8]) -> bool {
+        match protocol_type {
+            0x0800 => self
```

Comment:
There's probably a cleaner way to switch on the table type without duplicating the method calls every time.

Reply:
Yeeeeeeeep